### ISSUE TITLE
Outline Panel for Markdown Files erroneously shows keys of JSON as entries in the outline

### ISSUE THREAD

**Issue by rowillia** [zed-industries/zed#15122]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Open up `docs/src/languages/go.md` in the Zed repo in Zed
2. Open up the outline panel

Also: Note that I would expect the outline view to follow the same hierarchy as the document (e.g. `Inline Hints` would be nested under `Go`)

### Environment

Zed: v0.145.1 (Zed)
OS: macOS 14.5.0
Memory: 32 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="1285" alt="Screenshot 2024-07-24 at 4 57 06 PM" src="https://github.com/user-attachments/assets/0f14cd49-f5d3-45df-9073-11f034c58715">

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by jansol** [comment]
The JSON keys showing up actually is the correct behaviour I believe -- there is valid JSON embedded in the Markdown document which gets picked up by the markdown tree-sitter embedding rules. However, comparing to the outline of zed's settings file, the hierarchy of JSON keys looks broken here. "hints" should be at the top and the rest until "rangeVariableTypes" should be indented under it..

**Comment by JosephTLyons** [comment]
I agree with @jansol - Zed is pretty aware of code embedded in markdown, so it treats it as such. I'll leave this issue open regardless because of issues mentioned by Jansol.

cc: @SomeoneToIgnore

**Comment by opsnull** [comment]
When I write or read technical articles in Markdown, I usually use the outline panel to view the article's titles and subtitles, which allows me to quickly jump to corresponding sections. However, Zed currently displays code block content in the outline panel, which makes quick navigation by headings basically unusable. This is particularly inconvenient for longer documents that contain many code blocks. I suggest providing a configuration option to control whether parsed code blocks are displayed in the outline panel. Thank you!

**Comment by opsnull** [comment]
@SomeoneToIgnore thank you!
